ui/segmented-bar.reel/segmented-bar.js

```diff
@@ -0,0 +1,259 @@
+/**
+ * @module montage/ui/segmented-bar.reel
+ * @requires montage/ui/base/abstract-button.AbstractButton
+ * @requires montage/core/tree-controller.TreeController
+ */
+var AbstractButton = require("ui/base/abstract-button").AbstractButton;
+
+/**
+ * @class SegmentedBar
+ * @classdesc
+ * A segmented bar.
+ * Reference design: OS X storage space visualization found in iTunes or About This Mac
+ * Inheriting from AbstractButton because SegmentedBar needs its PressComposer behavior
+ *
+ * @extends AbstractButton
+ *
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Flexible_boxes}
+ * for FlexBox usage
+ */
+exports.SegmentedBar = AbstractButton.specialize( /** @lends SegmentedBar.prototype # */ {
```

Comment:
> BUG (I think): If you set a dataController value (legacy name: contentController :-), then let the segmented bar draw itself, then set a data value (legacy name: content), I don't think the segmented bar will correctly update the bindings to the repetition so the new data is displayed (because the bindings are set in templateDidLoad, which is called once and then never again). The most robust way to fix this is to ensure the segmented bar always has an RangeController, either taken or derived from the dataController or derived from the data. The repetition can then have a binding to this arrayController defined in the serialization and the binding doesn't need to change. Setters for data and dataController will ensure any changes get reflected in a correct _rangeController, which will update the repetition. Because you'll have better separation of concerns, the code for this might actually be smaller than the current version. No _data and _dataController properties will be necessary, and It's likely no needsDraw = 1 will be needed in the data and dataController setters because the binding will cause changes to _rangeController to propagate correctly to the repetition and force a redraw there. A test should be added for a dataController -> data change and vice versa.

Refactored out per discussion.